src/modules/vtol_att_control/standard.cpp

```diff
@@ -185,7 +185,20 @@ void Standard::update_vtol_state()
 
 			if (minimum_trans_time_elapsed) {
 				if (airspeed_triggers_transition) {
-					transition_to_fw = _airspeed_validated->calibrated_airspeed_m_s >= _params->transition_airspeed;
+
+					// The logic that will prevent spikes in airspeed measurements to trigger the transition
+					if ((_timestamp_transition_speed_detected == 0) &&
+					    (_airspeed_validated->calibrated_airspeed_m_s >= _params->transition_airspeed)) {
+						_timestamp_transition_speed_detected = hrt_absolute_time();
+
+					} else if (_airspeed_validated->calibrated_airspeed_m_s < _params->transition_airspeed) {
+						_timestamp_transition_speed_detected = 0;
+					}
+
+					if ((_timestamp_transition_speed_detected != 0) &&
```

Comment:
`_timestamp_transition_speed_detected` should be reset to 0 before the transition starts (in case of multiple transitions in the same flight. Otherwise there is a (very) small chance that a spike on the first sample considered will trigger transition complete.

Reply:
True. Fix pushed.